source/client-side-operations-timeout/tests/pending-response.yml

```
@@ -0,0 +1,349 @@
+          isTimeoutError: true
+      # Execute a final operation to drain the buffer.
+      - name: find
+        object: *collection
+        arguments:
+          filter: {_id: 1}
+    expectEvents:
+      - client: *client
+        events:
+          - commandFailedEvent:
+              commandName: insert
+          - commandSucceededEvent:
+              commandName: find
+      - client: *client
+        eventType: cmap
+        events:
+          - connectionCheckedOutEvent: {}
+          - connectionCheckedInEvent: {} # Insert fails.
+          - connectionPendingResponseStarted: {} # First find fails
+          - connectionPendingResponseFailed:
+              reason: timeout
+          - connectionPendingResponseStarted: {} # Second find drains the buffer.
+          - connectionPendingResponseSucceeded: {}
+          - connectionCheckedOutEvent: {}
+          - connectionCheckedInEvent: {} # Second find succeeds.
+    # If the connection is closed server-side while draining the response, the
+    # driver must retry with a different connection.
+  - description: "read op retries when connection closes server-side while draining response"
+    operations:
+      # Create a failpoint to block the first and second operation.
+      - name: failPoint
+        object: testRunner
+        arguments:
+          client: *failPointClient
+          failPoint:
+            configureFailPoint: failCommand
+            mode: {times: 1}
+            data:
+              failCommands: ["insert"]
+              blockConnection: true
+              blockTimeMS: 500
+              closeConnection: true
+      # Execute operation with timeout less than block time.
+      - name: insertOne
+        object: *collection
+        arguments:
+          timeoutMS: 50
+          document: {_id: 3, x: 1}
+        expectError:
+          isTimeoutError: true
+      - name: wait
+        object: testRunner
+        arguments:
+          ms: 500
+      # Execute a subsequent operation with a timeout less than the block time.
+      - name: find
+        object: *collection
+        arguments:
+          timeoutMS: 50
+          filter: {_id: 1}
+    expectEvents:
+      - client: *client
+        events:
+          - commandFailedEvent:
+              commandName: insert
+          - commandSucceededEvent:
+              commandName: find
+      - client: *client
+        eventType: cmap
+        events:
+          - connectionCheckedOutEvent: {}
+          - connectionCheckedInEvent: {} # Insert fails.
+          - connectionPendingResponseStarted: {} # Pending read fails on first find
+          - connectionPendingResponseFailed:
+              reason: error
+          - connectionClosedEvent:
+              reason: error
+          - connectionCheckedOutEvent: {}
+          - connectionCheckedInEvent: {} # Find finishes.
+    # If the connection is closed server-side while draining the response, the
+    # driver must retry with a different connection.
```

```suggestion
    # If the connection is closed server-side while draining the response, the
    # driver must retry with a different connection (write operation)
```

ditto for the test above, but with `read`.  Just reading the comments I thought these were duplicate tests.